server/src/main/java/com/cloud/server/ManagementServerImpl.java

```diff
@@ -3286,13 +3294,29 @@ protected void runInContext() {
         }
     }
 
+    private void cleanupDownloadUrls(VMInstanceVO systemVm) {
+        // clean download URLs when destroying ssvm
+        // clean only the volumes and templates of the zone
+        // to which ssvm belongs to
+        for (VolumeDataStoreVO volume :_volumeStoreDao.listVolumeDownloadUrlsByZoneId(systemVm.getDataCenterId())) {
```

Comment:
@ravening What if when one of the SSVM is destroyed (multiple SSVMs are created for a zone), which is created to meet the storage command sessions.

Reply:
then the entries will be cleared.

Reply:
> then the entries will be cleared.

ok @ravening, so all the zone url entries are deleted irrespective of the SSVM ip in the url, and a new url is generated whenever user/operator requests download. am i correct?

Reply:
@sureshanaparti yes